### PR TITLE
internal/v4: shorter timeout for debug/status

### PR DESCRIPTION
**Pull request by rogpeppe**
We also update a bunch of error messages to try to avoid the dreaded "EOF"
on mongo connection timeout.

### CODE REVIEW

File: internal/v4/status.go [hunk 1]
```diff
@@ -23,95 +24,105 @@ import (
 func (h *Handler) serveDebugStatus(_ http.Header, req *http.Request) (interface{}, error) {
 	store := h.pool.Store()
 	defer store.Close()
+	store.SetReconnectTimeout(500 * time.Millisecond)
```
Comment (urosj) [inline internal/v4/status.go:27]: Still not sure about this, but we can increase if it proves to be problematic.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/898/
Test PASSed.

**Comment by urosj** [comment]
:+1: on code, but would like run it as well

**Comment by urosj** [comment]
:+1: on QA

**Comment by urosj** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jrwren** [comment]
:+1: after the fact :)

**Pull request merged**